experimental/rocm/tracing.c

```diff
@@ -128,6 +157,23 @@ iree_status_t iree_hal_rocm_tracing_context_allocate(
         timestamp_period);
   }
 
+  context->smutrace_enabled = false;
+  void (*dl) = dlopen(SMUTRACE_DLL, RTLD_LAZY);
+  if (dl) {
+    context->f_smuDumpInit = (SmuDumpInitFunc) dlsym(dl, "smuDumpInit");
+    context->f_smuDumpEnd = (SmuDumpEndFunc) dlsym(dl, "smuDumpEnd");
+    context->f_smuDumpOnce = (SmuDumpOnceFunc) dlsym(dl, "smuDumpOnce");
+    context->f_regDumpOnce = (RegDumpOnceFunc) dlsym(dl, "regDumpOnce");
+    context->f_sviDumpOnce = (SviDumpOnceFunc) dlsym(dl, "sviDumpOnce");
+    context->f_smuGetTraceRate = (SmuGetTraceRate) dlsym(dl, "getSmuVariablesCaptureRate");
+    context->f_regGetTraceRate = (RegGetTraceRate) dlsym(dl, "getRegisterExpressionCaptureRate");
+    context->smutrace_enabled = (context->f_smuDumpInit && context->f_smuDumpEnd && context->f_smuDumpOnce &&
+                        context->f_smuGetTraceRate && context->f_regGetTraceRate && context->f_regDumpOnce  && context->f_sviDumpOnce  &&
+                        context->f_smuDumpInit(iree_smutrace_callback));
```

Comment:
In my own experiments I put hardware event collection code in Tracy itself: https://github.com/wolfpld/tracy/compare/master...ScottTodd:tracy:rocm-smi

For code that is independent of the application running, we should aim to integrate there instead IMO.

One thing that I ran into though was that the hardware event collection in Tracy seemed to continue past when the profiled application (e.g. IREE) finished. In my case I was sampling at a frequency higher than Tracy could ingest, so that led to a runaway where it would never actually finish...